x/superfluid/keeper/stake_test.go

```diff
@@ -663,334 +659,244 @@ func (suite *KeeperTestSuite) TestSuperfluidUnbondLock() {
-// 			[]assetTwap{{"gamm/pool/1", sdk.NewDec(0)}},
-// 			[]assetTwap{},
-// 			[]int64{0},
-// 		},
-// 		{
-// 			"refresh case from zero to non-zero",
-// 			[]stakingtypes.BondStatus{stakingtypes.Bonded},
-// 			[]superfluidDelegation{{0, 0, "gamm/pool/1", 1000000}},
-// 			[]assetTwap{{"gamm/pool/1", sdk.NewDec(0)}},
-// 			[]assetTwap{{"gamm/pool/1", sdk.NewDec(10)}},
-// 			[]int64{0},
-// 		},
-// 		{
-// 			"dust price twap check",
-// 			[]stakingtypes.BondStatus{stakingtypes.Bonded},
-// 			[]superfluidDelegation{{0, 0, "gamm/pool/1", 1000000}},
-// 			[]assetTwap{{"gamm/pool/1", sdk.NewDecWithPrec(1, 10)}}, // 10^-10
-// 			[]assetTwap{},
-// 			[]int64{0},
-// 		},
-// 		{
-// 			"refresh case from dust to non-dust",
-// 			[]stakingtypes.BondStatus{stakingtypes.Bonded},
-// 			[]superfluidDelegation{{0, 0, "gamm/pool/1", 1000000}},
-// 			[]assetTwap{{"gamm/pool/1", sdk.NewDecWithPrec(1, 10)}}, // 10^-10
-// 			[]assetTwap{{"gamm/pool/1", sdk.NewDec(10)}},
-// 			[]int64{0},
-// 		},
-// 	}
-
-// 	for _, tc := range testCases {
-// 		tc := tc
-// 		suite.Run(tc.name, func() {
-// 			suite.SetupTest()
-
-// params := suite.App.SuperfluidKeeper.GetParams(suite.Ctx)
-// bondDenom := suite.App.StakingKeeper.BondDenom(suite.Ctx)
-
-// 			// Generate delegator addresses
-// 			delAddrs := CreateRandomAccounts(tc.delegatorNumber)
-
-// 			// setup validators
-// 			valAddrs := suite.SetupValidators(tc.validatorStats)
-// 			// setup superfluid delegations
-// 			intermediaryAccs, locks := suite.SetupSuperfluidDelegations(delAddrs, valAddrs, tc.superDelegations)
-// 			suite.checkIntermediaryAccountDelegations(intermediaryAccs)
-// 			intermediaryDels := []sdk.Dec{}
-
-// 			for _, intAccIndex := range tc.checkAccIndexes {
-// 				expAcc := intermediaryAccs[intAccIndex]
-// 				valAddr, err := sdk.ValAddressFromBech32(expAcc.ValAddr)
-// 				suite.Require().NoError(err)
-
-// 				// check delegation from intermediary account to validator
-// 				delegation, found := suite.App.StakingKeeper.GetDelegation(suite.Ctx, expAcc.GetAccAddress(), valAddr)
-// 				fmt.Println("====delegation share")
-// 				fmt.Println(delegation.Shares)
-// 				suite.Require().True(found)
-// 				intermediaryDels = append(intermediaryDels, delegation.Shares)
-// 			}
-
-// 			// twap price change before refresh
-// 			twapByDenom := make(map[string]sdk.Dec)
-// 			for _, twap := range tc.roundOneTwaps {
-// 				twapByDenom[twap.denom] = twap.price
-// 				suite.App.SuperfluidKeeper.SetOsmoEquivalentMultiplier(suite.Ctx, 2, twap.denom, twap.price)
-// 			}
+func (suite *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
+	testCases := []struct {
+		name                string
+		validatorStats      []stakingtypes.BondStatus
+		delegatorNumber     int
+		superDelegations    []superfluidDelegation
+		roundOneMultipliers []osmoEquivalentMultiplier
+		roundTwoMultipliers []osmoEquivalentMultiplier
+		checkAccIndexes     []int64
+	}{
```

Comment:
I don't understand what this test is doing. It seems like it applies two refreshes for osmo equivalent TWAPs, and then at the end checks that every account index provided has 0 osmo in it?

How is this checking that the refresh operation applied as expected?

Reply:
It checks if intermediary account does not have balance after refresh operation.
And it is checking if the operation is applied by checking the delegation amount change.

```
targetDelegation := intermediaryDels[index].Mul(multiplier).Quo(originMultiplier)
```

Here `targetDelegation` is expected delegation after refresh.

Reply:
Ohh I see, I very much misunderstood this on first read. Thanks for updating the name